src/swsssdk/configdb.py

```diff
@@ -151,16 +156,32 @@ def typed_to_raw(self, typed_data):
                 raw_data[key] = str(value)
         return raw_data
 
+    # Note: we could not use a class variable for KEY_SEPARATOR, but original dependent code is using
+    # these static functions. So we implement both static and instance functions with the same name.
+    # The static function will behave according to ConfigDB separators.
     @staticmethod
     def serialize_key(key):
         if type(key) is tuple:
-            return ConfigDBConnector.KEY_SEPARATOR.join(key)
+            return '|'.join(key)
```

Comment:
We should not hardcode separator, but use the one obtained upon db_connect,

Reply:
This is static method, which already used by applications, and they are assuming ConfigDB separator.
Below is an instance method as you expected, and with a trick, I made them the same name.

---
In reply to: [549809761](https://github.com/Azure/sonic-py-swsssdk/pull/99#discussion_r549809761) [](ancestors = 549809761)